### PR TITLE
Update application logic

### PR DESCRIPTION
**Pull request by atstoyanov**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

**Review by techsmyth — COMMENTED** [review]
First pass at testing has this working nicely.

Realized when testing it that there is one case missing: user clicks apply on a Challenge and they already have an ecoverse application that is pending. In that case the popup should state the text below and then I think bring the user to the Challenge application form:
"To join 'tesats' you must join 'Default Ecoverse' ecoverse first. There is already an application pending for that 'Default Ecoverse", so you can also submit a parallel application for 'tesats'."

**Review by techsmyth — CHANGES_REQUESTED** [review]
Nice that the logic is there now for the challenge application once the ecoverse applicaiton is pending.

However the challenge does not show "application pending" after submitting the application.

steps:
- apply to join ecoverse
- apply to join child challenge
Apply button should not be still offered as an option to the user.

**Review by techsmyth — APPROVED** [review]
Working nicely now, thanks

File: src/components/composite/common/ApplicationButton/ApplicationButton.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { Dialog, DialogContent } from '@material-ui/core';
+import React, { FC, useMemo, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
```
Comment (hero101) [inline src/components/composite/common/ApplicationButton/ApplicationButton.tsx:4]: rename `Link` to `RouterLink` to avoid any future breaking changes.

**Review by hero101 — CHANGES_REQUESTED** [review]

**Review by hero101 — APPROVED** [review]

**Pull request merged**